[PORT] Port "Singulostation content"

KubeRoot opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Ports the following PR:
- https://github.com/SinguloStation13/SinguloStation-Blue/pull/1

See description in original PR for details

This PR has not been tested. Please either test the PR or wait until I get around to it.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->

This PR has **not** been tested.

<details>
<summary>Screenshots&Videos</summary>

N/A

</details>

## Changelog
:cl: KubeRoot, TheOneAndMightyRed, Urumasi, MaltVinegar, tgstation devs (tralezab), WhiteSands devs (AsciiSquid, MarkSuckerberg, Superyodeler)
add: Deepcore mining
add: CryoStation
add: ConstructionStation
add: Hardsuit and material crates to cargo
add: Singulostation cryogenic sleepers
add: Singulostation title card
add: Advanced multitool
tweak: Cable coil and construction bag capacity buffs
add: Transit tube dispensers
fix: :goberserk: (https://github.com/SinguloStation13/SinguloStation-Blue/pull/87)
add: Tripping
add: New dusting animation
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

tralezab committed:
Transit tube dispensers for JOHN GOBBEL on DISCORD.GG (#46344)

* relic file

* Revert "Merge branch 'master' of https://github.com/tralezab/tgstation"

This reverts commit 826916d87b421dbb0ebcfce7fad96e456b6d3674, reversing
changes made to a0f476d8746fa0d78f8a4b1363bd95fd265c2479.

* transit dispenser

* that should be all

* sprites

* true falses, dmi fixes

AsciiSquid committed:
Deepcore (#8)

* init: core machines and network datum

* get it compiling

* Added HUB

* realize I don't need component functions

* drill go burr

* full-featured drills

* barebones functions

* Ore veins no longer block the weather

* deepcore HUB

* last minute fix

* pinpoint skeleton

* adds to proc gen

* Mapgen fix and advanced pinpointers

* deepcore capsules

* thank you for the sprites zeta

* Added Ascii's stuff

* Re-add original scanner

best to keep it till we fully implement dcm more thoroughly

* Update waspstation/code/modules/cargo/packs.dm

* Update waspstation/code/modules/cargo/packs.dm

* Deep QoL

Co-authored-by: Mark Suckerberg <tommjd54321@outlook.com>
Co-authored-by: Yodeler <superjacob50@yahoo.com>

AsciiSquid committed:
patching things up (#18)

AsciiSquid committed:
Deepcore corrections (#19)

* init: core machines and network datum

* get it compiling

* Added HUB

* realize I don't need component functions

* drill go burr

* full-featured drills

* barebones functions

* Ore veins no longer block the weather

* deepcore HUB

* last minute fix

* pinpoint skeleton

* adds to proc gen

* Mapgen fix and advanced pinpointers

* deepcore capsules

* thank you for the sprites zeta

* Added Ascii's stuff

* Re-add original scanner

best to keep it till we fully implement dcm more thoroughly

* Update waspstation/code/modules/cargo/packs.dm

* Update waspstation/code/modules/cargo/packs.dm

* Deep QoL

* refactor DCM to mat containers

* fix dme being rude

* move to material containers

* Adds ores everywhere

* Comment lowmemmode

Knew I was forgetting something

Co-authored-by: Mark Suckerberg <tommjd54321@outlook.com>
Co-authored-by: Yodeler <superjacob50@yahoo.com>

AsciiSquid committed:
Deepcore ore prefabs (#20)

* init: core machines and network datum

* get it compiling

* Added HUB

* realize I don't need component functions

* drill go burr

* full-featured drills

* barebones functions

* Ore veins no longer block the weather

* deepcore HUB

* last minute fix

* pinpoint skeleton

* adds to proc gen

* Mapgen fix and advanced pinpointers

* deepcore capsules

* thank you for the sprites zeta

* Added Ascii's stuff

* Re-add original scanner

best to keep it till we fully implement dcm more thoroughly

* Update waspstation/code/modules/cargo/packs.dm

* Update waspstation/code/modules/cargo/packs.dm

* Deep QoL

* refactor DCM to mat containers

* fix dme being rude

* move to material containers

* Adds ores everywhere

* Comment lowmemmode

Knew I was forgetting something

* Ore vein prefabs

* typo

Co-authored-by: Mark Suckerberg <tommjd54321@outlook.com>
Co-authored-by: Yodeler <superjacob50@yahoo.com>

Tulenich commented:
didn't test it yet, gonna do that now
hope it won't crash because 144 errors appeared out of nowhere and i don't know code

Tulenich commented:
wait, i mean 26, not 144

Tulenich commented:
shit it crashed
i'll try again later

Tulenich commented:
I just noticed I made an error in _basemap.dm, I will need to fix that

Tulenich commented:
fuck it theredandmightyone is already doing it

Redory committed:
Adds CryoStation. (#4)

* Addes CryoStation

Redory committed:
Fixed haswallmount (#6)

MaltVinegar committed:
Urumasi's Improved dust animation (#12)

* Initial changes(doesn't work)

Co-Authored-By: Urumasi <8294011+Urumasi@users.noreply.github.com>

* Update death.dm

Co-Authored-By: Urumasi <8294011+Urumasi@users.noreply.github.com>

* finally works

* Modularization

Co-authored-by: Urumasi <8294011+Urumasi@users.noreply.github.com>

MaltVinegar committed:
Increases the amount of fuel used per cycle by 20 times (#15)

MaltVinegar committed:
Should unfuck firelocks (#16)

MaltVinegar committed:
Adds tripping (#17)

* Adds tripping

* Sorry for the web-edit

Redory committed:
Cryogenic freezers change (#22)

* cryopod console constructible

* Major changes to cryopod.dm - godmode now protects you more - new status effect for cryopods

* Major changes to cryopod.dm - tgui for cryopod oversight console - silicon compatibility

* Minor bugfixes in cryopod.dm - added cryopods to techweb - added auxilliary latejoin to maps - moved new icon to singulostation folder

* Added an edit comment

* Changed up the x_defense.dm files in regard to ex_act and emp_act

* added disclaimer to not ghost, tidied up chat messages upon entering a cryopod

MaltVinegar committed:
Makes walls upgradable into rwalls (#20)

* Update walls.dm

* Fixes bug

MaltVinegar committed:
Singulostation title card by Kube Root (#26)

* title card

* Animated title card

* Revert "Animated title card"

This reverts commit 857d81882cffefe9218bb2e70d6184b9865215c0.

* Animated title card with crediting

Co-Authored-By: KubeRoot <6917698+kuberoot@users.noreply.github.com>

Co-authored-by: KubeRoot <6917698+kuberoot@users.noreply.github.com>

MaltVinegar committed:
Plastic flap and holofirelock nerfs (#19)

* Holofirelock nerf

* Nerfs firedoors and plastic flaps

* Firedoor nerf didn't work...

* commend changes

* changes iscarbon check to ishuman check

Co-authored-by: TheOneAndMightyRed <41807890+TheOneAndMightyRed@users.noreply.github.com>

* Holofirelock buffed slightly and change to examine text

* Commend changes

Co-authored-by: TheOneAndMightyRed <41807890+TheOneAndMightyRed@users.noreply.github.com>

MaltVinegar committed:
Config update (#21)

KubeRoot committed:
Buff construction bags (#28)

* Make construction bags able to hold more, and hold sheets, rods and floor tiles

* Lower max_combined_w_class of construction bags to 400

Co-authored-by: MaltVinegar <69946338+MaltVinegar@users.noreply.github.com>

Co-authored-by: MaltVinegar <69946338+MaltVinegar@users.noreply.github.com>

KubeRoot committed:
Add advanced multitool (#36)

* Add advanced multitool

* Linter seems to want trailing newlines

* Make variable names more descriptive

* Apply review changes, add some comments

* Typecasts

* Whoops, all /

* Add codersprite (also dme rearranged by DreamMaker)

* Simplify code

KubeRoot committed:
Path of least resistance - teleport to location first, try to put into sleeper second in case that fails, and remove destroyed sleepers from latejoin locations (#39)

KubeRoot committed:
Make departmental protolathes and techfabs printable (#38)

* Make departmental protolathes and techfabs printable, and make circuit imprinter boards departmental (departmental-less)

* Fix wrong departments in designs, move to industrial engineering, add missing comment

* T R A I L I N G   N E W L I N E

* AUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENT

AUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENTAUTOINDENT

Co-authored-by: Urumasi <Urumasi@email.cz>

Co-authored-by: Urumasi <Urumasi@email.cz>

Terra1 committed:
Merge pull request #40 from KubeRoot/roundstart-construction-bags

Add construction bags to Robco vending machines, available roundstart

Terra1 committed:
Merge pull request #42 from KubeRoot/copy_airlock_electronics

Add airlock electronics access copying

Terra1 committed:
Merge pull request #30 from Urumasi/typo

Merge requested by maintainer.

Terra1 committed:
Merge pull request #46 from Urumasi/firedooroffset

Merge requested on Discord.

Urumasi committed:
Make deepcore pinpointers only locate ores on the same Z-level (#43)

* Only taget same Z

* Add singulo comment

KubeRoot committed:
Fix radial menu transparency (#79)

* radial menus ignore the anchor's alpha, color and transform now. (#54870)

* Add singulostation comment

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

KubeRoot committed:
Make silo link usable by all constructors (#53)

* Add right-click verb available across all constructors to toggle silo link

* Move RCD toggle_silo_link proc to constructor base class
Make use of toggle_silo_link in verb
Add silo link entry to plumbing construction radial menu

Urumasi committed:
Make walking disable tripping (#86)

KubeRoot committed:
Make conveyor switches togglable between one-way and two-way (#81)

Make conveyor switches persist state when picked up and placed back down

Urumasi committed:
:goberserk: (#87)

MaltVinegar committed:
Increases RCL's capacity (#88)

* :train:

* :train2:

KubeRoot committed:
Make dispenser stations buildable (#82)

Fix dispenser station sprite issues

KubeRoot committed:
Add some WS hardsuit and material crates

KubeRoot committed:
Move waspstation folder to whitesands, add simple readme

KubeRoot committed:
Add cryogenic freezer permasleep

KubeRoot committed:
Whoops, the gitignore change was supposed to be local only

KubeRoot committed:
Lower time before dangerous events from 80 hours to 5 hours

KubeRoot committed:
Disable autotransfer

KubeRoot committed:
Increase cable coil stack size 10x (#51)

Allow for cable coil recoloring

KubeRoot committed:
Fix turbine runtime and RPM gain when intake is solid turf (for example wall) (#85)

KubeRoot committed:
Constructionstation and cargo shuttle (#5)

* Constructionstation and cargo shuttle

* Config update

* Mapmerge

* Update _basemap.dm

* f

* Really? This is what the checks were crying about?

* I have brain damage

* Fixes some things

* improvements

* More Constructionstation improvements

* Finishing touches

* fixes some areas

* area fixes

* Forgot what I came here for

* Fixes issues pointed out by redory

* More fixes

* More fixes

* Kube roots fixes

KubeRoot committed:
Add missing file

KubeRoot committed:
Remove misplaced SIGNAL_HANDLER

KubeRoot committed:
CI Trigger

KubeRoot committed:
Fix cryostation cargo shuttle

KubeRoot committed:
Remove remaining beestation title screens

KubeRoot committed:
Add CryoStation to ALL_MAPS

KubeRoot committed:
Fix cryostation

KubeRoot committed:
Reenable AI due to latejoin breaking

KubeRoot committed:
Fix cable recolor var access - item_color to cable_color

KubeRoot committed:
Remove force cryo punishment

KubeRoot committed:
Fix typepath in science hardsuit crate

Laboredih123 committed:
Fix atmos layers on maps and shuttle DMM location

Co-authored-by: LaboredGaming <62359847+LaboredGaming@users.noreply.github.com>

MaltVinegar reviewed (approved):
Everything looks good :+1:
Holding off on merging until I can get the PRMirror bot to work

KubeRoot committed:
Add trailing newlines
